### DISCUSSION TITLE
More clarity about patterns in config files

### DISCUSSION THREAD

**Discussion by arcticmatt** [vercel/next.js#25616] [Help]
From https://nextjs.org/docs/api-reference/next.config.js/headers

> source: A pattern that matches each incoming pathname (excluding querystring).

From https://vercel.com/docs/configuration#project/headers

> source: A pattern that matches each incoming pathname (excluding querystring).

These docs are confusing. What patterns? Regex? Glob?

AFAICT, the former seems like a combination of regex and glob. The latter uses regex (I think?). It would be nice if the docs made this more clear.

**Comment by arcticmatt** [comment]
There is some documentation about the `vercel.json`pattern! https://vercel.com/docs/platform/frequently-asked-questions#invalid-route-source-pattern

I found it after specifying an incorrect pattern. It would be great if this was in the original docs (https://nextjs.org/docs/api-reference/next.config.js/headers)

**Comment by lasfito** [comment]
One year has passed and the docs havent changed. 
Just stating "patter" is confusing!

**Comment by koodistrom** [comment]
Two  years have passed and the docs haven't changed.
Just stating "patter" is confusing!

Found some information about the pattern syntax in: 
https://vercel.com/docs/errors/error-list#invalid-route-source-pattern

The source property follows the syntax from [path-to-regexp](https://github.com/pillarjs/path-to-regexp/tree/v6.1.0)
, not the RegExp syntax.

**Reply by johnnygerard** [reply]
This seems to be correct.
Check out https://github.com/vercel/vercel/blob/main/packages/routing-utils/package.json#L23
